htmresearch/algorithms/column_pooler.py

```diff
@@ -273,25 +282,198 @@ def _computeInferenceMode(self, feedforwardInput, lateralInputs):
         lateralInput, self.connectedPermanenceDistal)
       numActiveSegmentsByCell[overlaps >= self.activationThresholdDistal] += 1
 
-    # Activate some of the feedforward supported cells
-    minNumActiveCells = self.sdrSize / 2
-    chosenCells = self._chooseCells(feedforwardSupportedCells,
-                                    minNumActiveCells, numActiveSegmentsByCell)
+    chosenCells = []
+    minNumActiveCells =  30 #self.sdrSize / 2
+
+    # # # # Old method. Relies on the chooseCells function.
+    # chosenCells = self._chooseCells(feedforwardSupportedCells,
+    #                                 minNumActiveCells, numActiveSegmentsByCell)
+    # # If necessary, activate some of the previously active cells
+    # if len(chosenCells) < minNumActiveCells:
+    #   remaining = numpy.setdiff1d(prevActiveCells, feedforwardSupportedCells)
+    #   remaining = remaining[numActiveSegmentsByCell[remaining] > 0]
+    #
+    #   chosenCells = numpy.append(
+    #     chosenCells, self._chooseCells(remaining,
+    #                                    minNumActiveCells - len(chosenCells),
+    #                                    numActiveSegmentsByCell))
+
+
 
-    # If necessary, activate some of the previously active cells
+
+
+    # Simple, coherent method - but less good performance...
+
+    remainingFFSupportedCells = numpy.setdiff1d(feedforwardSupportedCells, chosenCells)
+    numActiveSegsForFFSuppCells = numActiveSegmentsByCell[remainingFFSupportedCells]
+
+    # First, activate the FF-supported cells that have the highest number of lateral active segments (this may be 0)
+    if len(remainingFFSupportedCells) == 0:
+        pass
+    else:
+        # This loop will select the FF-supported AND laterally-active cells, in
+        # order of descending lateral activation, until we exceed the
+        # minNumActiveCells quorum - but will exclude cells with 0 lateral
+        # active segments!
+        ttop = numpy.max(numActiveSegsForFFSuppCells)
+        if ttop > 0:
+            while True:
+                chosenCells = numpy.unique(numpy.append(chosenCells,
+                        remainingFFSupportedCells[numActiveSegsForFFSuppCells >= self.distalSegmentInhibitionFactor * ttop]))
```

Comment:
This should probably be `>`, not `>=`. For example, if this factor is `.5`, and if `ttop` is 2, then we probably want cells with 1 segment to be inhibited.

(I like that you changed the `distalSegmentInhibitionFactor` to its inverse. This math is much clearer.)